File: backend/app/models/money.py
from datetime import date, datetime
from decimal import Decimal
from typing import Literal

from pydantic import BaseModel, ConfigDict


class CurrencyOut(BaseModel):
    code: str
    exponent: int
    name: str
    symbol: str | None = None


class ExchangeRateOut(BaseModel):
    base_currency: str
    quote_currency: str
    rate: Decimal
    rate_date: date
    source: str
    fetched_at: datetime


class FXValuationOut(BaseModel):
    model_config = ConfigDict(from_attributes=True)

    base_currency: str
    amount_minor_base: int | None
    exchange_rate_to_base: Decimal | None
    exchange_rate_source: str | None
    exchange_rate_as_of: datetime | None
    fx_valuation_status: Literal["not_required", "provisional", "final", "missing"]

File: mobile/src/types/money.ts
export type CurrencyCode = "PHP" | "USD" | "JPY" | "KWD" | string;

export type CurrencyMeta = {
  code: CurrencyCode;
  exponent: number;
  name: string;
  symbol?: string | null;
};

export type ExchangeRate = {
  base_currency: CurrencyCode;
  quote_currency: CurrencyCode;
  rate: string;
  rate_date: string;
  source: string;
  fetched_at: string;
};

export type RatesResponse = {
  base_currency: CurrencyCode;
  currencies: CurrencyMeta[];
  rates: ExchangeRate[];
};

